### ISSUE TITLE
add support for 2FA/MFA

### ISSUE THREAD

**Issue by kplimack** [Netflix-Skunkworks/go-jira#86]
I use _duo_ for 2-factor-authentication and `gojira` doesnt seem to support it.  If i net the `password-source` to _pass_, I can see Duo's 2fa page returned, but I'm unsure how to provide it anything.

**Comment by coryb** [comment]
Yeah, sounds like go-jira would not support that setup currently.  Unfortunately I think you are on your own to solve this one since I wont have access to test your setup.  If you find a solution definitely send me a PR for it and I will get it merged in.

**Comment by coryb** [comment]
I have rewritten go-jira to use the session login api in Jira, not sure if that will work for you or if the 2FA/Duo login is still required for API access to your jira instance. Might want to give the latest version a try: https://github.com/Netflix-Skunkworks/go-jira#login-process-change

**Comment by coryb** [comment]
We have added api-token support which should work for any hosted jira on atlassian.not, not sure if this setup will work for you: https://github.com/Netflix-Skunkworks/go-jira#authentication
